server/routes/events.js

```diff
@@ -12,120 +13,447 @@ router.get('/ping', function(req, res, next) {
+  	util.throwError(400, "Missing 'ends_at' parameter.");
+  }
+  if (!req.body.quarter) {
+  	util.throwError(400, "Missing 'quarter' parameter.");
+  }
+
+  let category_ids = req.body.categories;
+  let category_values = [];
+  let company_ids = req.body.companies;
+	let company_values = [];
+	let host_ids = req.body.hosts;
+  let host_values = [];
+
+  if (category_ids) {
+  	category_ids.forEach(function(entry) {
+  		category_values.push({ event_id: req.body.event_id, category_id: entry });
+  	});
+  }
+
+  if (company_ids) {
+		company_ids.forEach(function(entry) {
+			company_values.push({ event_id: req.body.event_id, company_id: entry });
+		});
+	}
+
+  if (host_ids) {
+  	host_ids.forEach(function(entry) {
+	  	host_values.push({ event_id: req.body.event_id, host_id: entry });
+	  });
+	}
+
+  var events_query = knex('event').insert(values);
+  var category_query = knex('event_category').insert(category_values);
+  var company_query = knex('event_company').insert(company_values);
+  var host_query = knex('event_host').insert(host_values);
+
+	knex.transaction(function(trx) {
+		return events_query.transacting(trx)
+			.then(async function() {
+				await category_query.transacting(trx);
+				await company_query.transacting(trx);
+				await host_query.transacting(trx);
+				return trx.commit;
+			})
+	})
+	.then(function() {
+		res.send(util.message('Successfully inserted new event: ' + req.body.name));
+	})
+	.catch(err => { return next(err) });
+});
+
+// Updated a single event by event_id
+router.put('/:event_id', function(req, res, next) {
+	const event_id = req.params.event_id;
+	let validReqForEvent = false;
+
+	let values = {
+  	name: req.body.name, 
+  	starts_at: req.body.starts_at, 
+  	ends_at: req.body.ends_at,
+  	quarter: req.body.quarter,
+  	location_id: req.body.location_id,
+  	description: req.body.description,
+  	fb_event: req.body.fb_event,
+  	picture: req.body.picture,
+  	is_featured: req.body.is_featured
+  };
+
+  let category_ids = req.body.categories;
+	let category_values = [];
+  let company_ids = req.body.companies;
+	let company_values = [];
+  let host_ids = req.body.hosts;
+  let host_values = [];
+
+  if (category_ids){
+	  category_ids.forEach(function(entry) {
+	  	category_values.push({ event_id: event_id, category_id: entry });
+	  });
+	}
+
+  if (company_ids){
+	  company_ids.forEach(function(entry) {
+	  	company_values.push({ event_id: event_id, company_id: entry });
+	  });
+	}
+
+  if (host_ids){
+	  host_ids.forEach(function(entry) {
+	  	host_values.push({ event_id: event_id, host_id: entry });
+	  });
+	}
+
+	let remove_category_ids = req.body.remove_categories;
+	let remove_company_ids = req.body.remove_companies;
+  let remove_host_ids = req.body.remove_hosts;
+
+  Object.keys(values).forEach(function(key) {
+  	if (!validReqForEvent && values[key]) {
+  		validReqForEvent = true;
+  	}
+	});
+
+	var event_query = knex('event').modify(function(queryBuilder) {
```

Comment:
Sorry a bit confused with this syntax, is this necessary/the simplest way to do this?